### PR TITLE
Untangle: Remove My Mailboxes menu from the classic view

### PR DESCRIPTION
**Pull request by arthur791004**
Fixes https://github.com/Automattic/dotcom-forge/issues/5310

## Proposed changes:

* This PR removes the My Mailboxes menu from the classic view

| Before | After |
| - | - |
| image | image |

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Head to Settings → Hosting Configuration.
* Scroll to the Admin interface style section, and select Classic style.
* Refresh the page if needed.
* Ensure that the My Mailboxes menu is no longer present in the sidebar

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php [hunk 1]
```diff
@@ -74,10 +74,6 @@ public function reregister_menu_items() {
 		$this->add_my_home_menu();
 		$this->remove_gutenberg_menu();
 
-		if ( ! get_option( 'wpcom_is_staging_site' ) ) {
```
Comment (taipeicoder) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php:78]: We still need to keep it for the default style 🥲

Comment (lupus2k) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php:78]: Yes, we don't want to change anything in the Default style. I suggest to use the same code we are using in other PRs:
```
if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
```

**Comment by arthur791004** [comment]
We have to wait for the product decision to see whether we can simply remove it from the menu.

**Comment by autumnfjeld** [comment]
@arthur791004  Mailboxes will remain and go in the wp.com menu item p1706560998383699/1706560885.265569-slack-C06DN6QQVAQ

**Comment by arthur791004** [comment]
@autumnfjeld Referring to https://github.com/Automattic/dotcom-forge/issues/5310#issuecomment-1920113580 and pbLl1t-205-p2, we still need to remove “My Mailboxes” as it's redundant and it can be accessed from “ Emails > Email Settings > View Mail” (See https://github.com/Automattic/dotcom-forge/issues/5397 for more details)

**Review by lsl — APPROVED** [review]

**Pull request merged**